tuner/tuner/candidate_gen.py

```diff
@@ -147,6 +147,50 @@ def get_default_output_dir() -> str:
     return "tuning_" + datetime.now().strftime("%Y_%m_%d_%H_%M")
 
 
+def determine_td_specs_to_link(
+    td_specs: list[ir.Module], warned_overlap_matchers: set[str]
+) -> tuple[list[ir.Module], set[str]]:
+    """
+    Determines which tuning specs should be linked based on matcher overlap.
+
+    Additionally, the function identifies any overlapping matchers between the
+    starter and current tuning specs that haven't yet triggered a warning.
+
+    Args:
+        td_specs: A list of 1 or 2 tuning spec modules. If two are provided, the first is
+                the starter spec.
+        warned_overlap_matchers: Set of matchers already warned about for overlaps.
+
+    Returns:
+         A tuple containing:
+        - A list of td specs to link (possibly excluding the starter spec).
+        - A set of overlapping matchers that haven't been warned about yet.
+    """
+
+    if len(td_specs) == 1:
+        # No starter td spec provided, nothing to merge.
+        return td_specs, set()
+
+    assert len(td_specs) == 2, "Expected exactly two TD specs (starter and current)"
+    starter_td_spec = td_specs[0]
+    current_td_spec = td_specs[1]
```

Comment:
nit: you can unpack it like this
```suggestion
    starter_td_spec, current_td_spec = td_specs
```